test/EFCore.PG.FunctionalTests/Query/NorthwindFunctionsQueryNpgsqlTest.cs

```diff
@@ -41,6 +41,18 @@ public override Task Where_math_log_new_base(bool async)
         public override Task Convert_ToString(bool async)
             => base.Convert_ToString(async);
 
+        [ConditionalTheory]
+        [MemberData(nameof(IsAsyncData))]
+        public virtual async Task Convert_ToDateTime(bool async)
+        {
+            await AssertQuery(
+                async,
+                ss => ss.Set<Order>().Where(o => o.OrderDate.HasValue && o.OrderDate == Convert.ToDateTime(o.OrderDate.ToString())),
```

Comment:
You should be able to omit the HasValue check, EF Core is null-safe.

Reply:
I could. But then the translated query would look like this:
```sql
WHERE (o."OrderDate" = CAST(CAST(o."OrderDate" AS text) AS timestamp without time zone)) OR (o."OrderDate" IS NULL)
```
And I don't like that OR so much. So I thought that using the .HasValue would be better. But I can change it if you prefer.

Reply:
Done.